File: Problem_Sets/LeetCode/Valid Parentheses/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        openBrackets = []
        closeBrackets = {')': '(', ']': '[', '}': '{'}
        for char in s:
            if char in closeBrackets:
                if openBrackets and openBrackets[-1] == closeBrackets[char]:
                    openBrackets.pop()
                else:
                    return False
            else:
                openBrackets.append(char)
        return True if not openBrackets else False

File: Problem_Sets/LeetCode/Valid Parentheses/ValidParentheses.cpp
#include <stack>

class Solution {
public:
    bool isValid(string s) {
        stack<char> openBrackets;
        for (char currentChar : s) {
            if (currentChar == '(' || currentChar == '{' || currentChar == '[') {
                openBrackets.push(currentChar);
            } 
            else 
            {
                switch(currentChar) {
                    case ')':
                        if (!openBrackets.empty() && openBrackets.top() == '(') {
                            openBrackets.pop();
                        }
                        else {
                            return false;
                        }
                        break;
                    case '}':
                        if (!openBrackets.empty() && openBrackets.top() == '{') {
                            openBrackets.pop();
                        }
                        else {
                            return false;
                        }
                        break;
                    case ']':
                        if (!openBrackets.empty() && openBrackets.top() == '[') {
                            openBrackets.pop();
                        }
                        else {
                            return false;
                        }
                        break;
                }
            }
        }
        if (!openBrackets.empty()) {
            return false;
        }
        return true;
    }
};